File: Frontend/testing-app/src/app/shared/interfaces/tests/TestInfo.ts
import { ProfileInfo } from "../Profile/ProfileInfo";
import { TestType } from "./TestType";

export interface TestInfo {
  Id: number,
  Name: string,
  Description: string,
  CreatedTime: Date,
  IsPublic: boolean,
  TestType: TestType,
  DurationInMinutes: number | null,
  Owner: ProfileInfo
}

File: Backend/TestRating/TestRating.Application/Contacts/Test/TestInfo.cs
using TestRating.Application.Contacts.Profile;

namespace TestRating.Application.Contacts.Test
{
    public class TestInfo
    {
        public long Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        public DateTime CreatedTime { get; set; }

        public bool IsPublic { get; set; }

        public string TestType { get; set; } = string.Empty;

        public double? DurationInMinutes { get; set; }

        public required BaseProfileResponse Owner { get; set; }
    }
}
